Crystal broken on darwin/macOS (not just 0.32.x)

### Describe the bug
From macOS/darwin I cannot build the `crystal` package.

### Steps To Reproduce
1. Get a Mac or find someone with a Mac :D
2. Ensure you are on the `nixpkgs-unstable` channel.
3. Execute `nix-shell -p crystal`.
4. Observe the package being broken.

Here's my output:
```sh
$ nix-shell -p crystal
error: Package ‘crystal-0.32.1’ in /nix/store/z18rhsp1qlh56kgrfjzv9k5ad1hy9brz-nixpkgs-21.11pre303315.16105403bdd/nixpkgs/pkgs/development/compilers/crystal/default.nix:212 is marked as broken, refusing to evaluate.

a) To temporarily allow broken packages, you can use an environment variable
   for a single invocation of the nix tools.

     $ export NIXPKGS_ALLOW_BROKEN=1

b) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

(use '--show-trace' to show detailed location information)
```

### Expected behavior
I expect `crystal` 1.0 to install. That it seems to be installing 0.32.1 is surprising to me.

### Additional context
I have looked around for current and prior issues regarding this, but didn't see any. My apologies if I've missed something obvious in my notes below. My Nix knowledge is still in its infancy.

[This line](https://github.com/NixOS/nixpkgs/blob/3b6c3bee9174dfe56fd0e586449457467abe7116/pkgs/development/compilers/crystal/default.nix#L279) suggests to me that 1.0 is installed by default, but [this line](https://github.com/NixOS/nixpkgs/blob/3b6c3bee9174dfe56fd0e586449457467abe7116/pkgs/development/compilers/crystal/default.nix#L218) seems to be what is being executed (0.32 being unable to install on macOS). [This comment](https://github.com/NixOS/nixpkgs/blob/3b6c3bee9174dfe56fd0e586449457467abe7116/pkgs/development/compilers/crystal/default.nix#L28-L31) indicates some version chaining might be going on, but I might be grasping at straws.

My understanding is the `binary` attribute is sort of being derived in a chained fashion, but the derivation is configured such that the most recent version provides the overrides. So I wouldn't expect there to be any problems with this package since I have not indicated to install 0.32.x.

I have also tried `nix-shell -p crystal_1_0`. It provides the same messages re: 0.32.1.

Running the commands again with `NIXPKGS_ALLOW_BROKEN=1` does start a build, but again it tries to build 0.32.1. This ultimately fails with the same reason listed in the [derivation's comment for breaking 0.32.x](https://github.com/NixOS/nixpkgs/blob/3b6c3bee9174dfe56fd0e586449457467abe7116/pkgs/development/compilers/crystal/default.nix#L217). Using `crystal_1_0` produces the same behavior.

### Notify maintainers
@david50407 @fabianhjr @manveru @peterhoeg 

@SuperSandro2000's [commit](https://github.com/NixOS/nixpkgs/commit/2ab8b31520fd5d2384e2a81b3bb358ffc1f7a747) has put in the relevant line, but I don't think this is the cause. Without it I'd still encounter a failure, just later.

### Metadata

```
$ nix-shell -p nix-info --run "nix-info -m"
these paths will be fetched (0.01 MiB download, 0.03 MiB unpacked):
  /nix/store/hsp7hzwkfqf5vx898a6xbqn0122axry2-DarwinTools-1
  /nix/store/j2n24ws0w3956cqq6c311ks42q22c75j-nix-info
copying path '/nix/store/hsp7hzwkfqf5vx898a6xbqn0122axry2-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/j2n24ws0w3956cqq6c311ks42q22c75j-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.4.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.10`
 - channels(logan): `"home-manager"`
 - channels(root): `"nixpkgs-21.11pre303315.16105403bdd, home-manager"`
 - nixpkgs: `/Users/logan/.nix-defexpr/channels/nixpkgs`
```

I'm taking best guess on the attributes.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: crystal shards oq
# a list of nixos modules affected by the problem
module:
```
### Appreciation

Thanks everyone for maintaining this project! Nix controls my entire local setup, and there's no going back to manual installs again :)

Thanks for the fast response, folks!

TIL about `gh` and `nixpkgs-review`. Slick!

There's quite a bit of output here, but the short of it I think is that now it breaks at 0_33.
``` sh
$ nixpkgs-review pr 131808
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/131808/head:refs/nixpkgs-review/1
remote: Enumerating objects: 391, done.
remote: Counting objects: 100% (336/336), done.
remote: Compressing objects: 100% (152/152), done.
remote: Total 391 (delta 228), reused 246 (delta 177), pack-reused 55
Receiving objects: 100% (391/391), 855.50 KiB | 7.31 MiB/s, done.
Resolving deltas: 100% (228/228), completed with 74 local objects.
From https://github.com/NixOS/nixpkgs
 * [new branch]              master                -> refs/nixpkgs-review/0
 * [new ref]                 refs/pull/131808/head -> refs/nixpkgs-review/1
$ git worktree add /Users/logan/.cache/nixpkgs-review/pr-131808/nixpkgs 3cc3160fe6cb3b85ac9b8f77b4c5d3b68f9c2962
Preparing worktree (detached HEAD 3cc3160fe6c)
Updating files: 100% (27197/27197), done.
HEAD is now at 3cc3160fe6c Merge pull request #131860 from fpletz/pkgs/jool-4.1.5
$ git merge --no-commit --no-ff 2d9674ecf5636d0be34b428d4f853eb71e36d0ff
Automatic merge went well; stopped before committing as requested
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
warning: ignoring the user-specified setting 'system', because it is a restricted setting and you are not a trusted user
$ nix --experimental-features nix-command build --no-link --keep-going -f /Users/logan/.cache/nixpkgs-review/pr-131808/build.nix
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
copying path '/nix/store/rbdh6qwh9b310mgy1j8xz1p9gxjc9lg4-install-shell-files' from 'https://cache.nixos.org'...
copying path '/nix/store/i2yanx31r9vs1mc9asp561818f265aq6-jq-1.6-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/6azv1j24pzr3s1si3mas4k4krw7habnh-readline-6.3p08-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/00kfi2iw9dvr92dxidxqd6wqdw86m3mw-source' from 'https://cache.nixos.org'...
copying path '/nix/store/0b3fhvz23sr3zd4kdkhmsa5zhypn0fsl-source' from 'https://cache.nixos.org'...
copying path '/nix/store/33p4k6y60p70xyf1jnd6wjz8x75r16lw-source' from 'https://cache.nixos.org'...
copying path '/nix/store/344nrb9j3ijynmdqml32x599cmwgj9jg-source' from 'https://cache.nixos.org'...
copying path '/nix/store/3z9jb2nhpw5n1xl3lxcpwwjwzayxddjl-source' from 'https://cache.nixos.org'...
copying path '/nix/store/4pxrhlwdwh5ai4lxhm5kmfbhsi3ljv6n-source' from 'https://cache.nixos.org'...
copying path '/nix/store/5k6bc017pcnysjgi3qkglq7v5rrsrx2v-source' from 'https://cache.nixos.org'...
copying path '/nix/store/7rdw2dlf261kh7h83ynkxl45cghr1w18-source' from 'https://cache.nixos.org'...
copying path '/nix/store/82mrg136a2z904crr9qqq7y3dpf25x5j-source' from 'https://cache.nixos.org'...
copying path '/nix/store/a7zqz0ymk0shixjgnp5g3w99mp9389m2-source' from 'https://cache.nixos.org'...
copying path '/nix/store/avmk9plx92qxhzggh9rci8cfhy76dhk7-source' from 'https://cache.nixos.org'...
copying path '/nix/store/bsmrpn35khgrwsprykq8yvyhcnclxhm7-source' from 'https://cache.nixos.org'...
copying path '/nix/store/cckyvddacggyfsn253anjq9pd6glhzgx-source' from 'https://cache.nixos.org'...
copying path '/nix/store/g7xcw9b2h7rnaxkhiv6cc7hbgrc217bx-source' from 'https://cache.nixos.org'...
copying path '/nix/store/inqap1yf1vq1pxvxn452kklpqipwqk5y-source' from 'https://cache.nixos.org'...
copying path '/nix/store/jbx8rfcsql90hc8xjrrb4pgwr7vszbm2-source' from 'https://cache.nixos.org'...
copying path '/nix/store/l78ig7f18abnv179rviyz383bvhk1p94-source' from 'https://cache.nixos.org'...
copying path '/nix/store/m03i2vw8s861a33h4kav1sd5glsvyj8w-source' from 'https://cache.nixos.org'...
copying path '/nix/store/m5hn50cknr366xd79rh6dhnhhbx6l4fm-source' from 'https://cache.nixos.org'...
copying path '/nix/store/pny099a2hy4hcwmmgxlhgajbpkzr9yqi-source' from 'https://cache.nixos.org'...
copying path '/nix/store/q1z2vjd8hankjbw59sxbkr9aaqwmm6lw-source' from 'https://cache.nixos.org'...
copying path '/nix/store/sdsl1i1rgvx8cj2l23iasmqkcgj180il-source' from 'https://cache.nixos.org'...
copying path '/nix/store/v2c1bjlgk6vcxcc1c30av5ascxqr0hdm-source' from 'https://cache.nixos.org'...
copying path '/nix/store/vzzy9s4vfk6llswbis85rdfwf9bfky7g-source' from 'https://cache.nixos.org'...
copying path '/nix/store/w6nad3hbav7bbs1gb6bq0z59vw2mm2vs-source' from 'https://cache.nixos.org'...
building '/nix/store/fni7j6mlgzay4mgnydfhjys2lp83nyxq-crystal-0.33.0-1-darwin-x86_64.tar.gz.drv'...
building '/nix/store/0pblaikfk6r06hy0hmffilibrq0cz3m5-crystal-lib.drv'...
building '/nix/store/zavaqv83lzyxqazv3zmcfph1fk52pmyi-crystal-binary-0.33.0.drv'...
building '/nix/store/1yc2c3bj70zp4qlmzgx4g02v8avyp586-crystal-lib.drv'...
building '/nix/store/my4dbgdf3xly9f7nmb91miyiimizavwi-crystal-0.33.0.drv'...
[1/4/25 built, 28 copied (6.4/6.4 MiB), 1.6 MiB DL] building crystal-0.33.0 (checkPhase): ld: warning: directory not found for option '-L/nix/store/qj6llsa4fjppph8fj0rhv9fqyp52z2f6 

builder for '/nix/store/my4dbgdf3xly9f7nmb91miyiimizavwi-crystal-0.33.0.drv' failed with exit code 2; last 10 log lines:
    Semantic: instance var doesn't infer type to be nilable if using self.class in call in assign
    Semantic: pointer allows using pointer with subclass
    Semantic: primitives types pointer of int
  
  Failures:
  
    1) Crystal::Command::FormatCommand formats stdin (bug + show-backtrace)
  
  Error running at_exit handler: Nil assertion failed
  make: *** [Makefile:89: compiler_spec] Error 1
cannot build derivation '/nix/store/pha43pd85jrz09dp89b8q01x73p0jpcb-crystal-0.34.0.drv': 1 dependencies couldn't be built
building '/nix/store/4n87nhmj3vadna53nz1qycs9990p97jk-crystal-lib.drv'...
cannot build derivation '/nix/store/jd5pdzd0h07kmbz71n72p4lwxs988x8c-crystal-0.35.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/5xwlhghzw0qrvnf0xy8m4p626pk9g6k4-crystal-0.36.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/mdrw66hv4n4qfr6ahq73ygjxlyqf4lg4-crystal-1.0.0.drv': 1 dependencies couldn't be built
building '/nix/store/ald0s5dx50cax7g73zi3b080dng5vm08-crystal-lib.drv'...
cannot build derivation '/nix/store/q61qk26jlwyl7v0qvgwsb4qa9shj07yl-lucky-cli-0.26.0.drv': 1 dependencies couldn't be built
building '/nix/store/dvbhz7j0ndc4qbz2r9n6022yq170vnly-crystal-lib.drv'...
building '/nix/store/j32h8gbf62kxnyr2xrzw59hxs6xzh98q-crystal-lib.drv'...
building '/nix/store/kysg194jcxkw8xg9b0kd6z5mk227n0d9-crystal-lib.drv'...
cannot build derivation '/nix/store/1y2g53kf9hbbx8hiba8yia3g6479lgbf-shards-0.11.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/p2wcvvml6fz61gklrag74jb06m5113ng-shards-0.14.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ip70p46b8qa26y672xrj941b6ls18iiw-ameba-0.14.3.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/r31j1lk7zbf40bv056v9rjqz1z579wka-crystal2nix-0.1.1.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/a9b6mvr1vkkrrvxjr34wj33nnwygrpp0-icr-unstable-2021-03-14.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/5yjr1i38y7nphgcl7al3p3f4zmrskgsc-mint-0.14.0.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/kkd4bf17vxxszavnjjbg4s39bi6544xh-oq-1.2.0.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/i9nhq9p2ir5f4nqkx72lvmn0jnjqd4a6-thicket-0.1.4.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/yaw12q07zif10wdmr7c8jvvbk3z7qnrc-scry-0.9.1.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/vlmf0pkf8fshclazwcl9h7nqy3faw5aa-review-shell.drv': 15 dependencies couldn't be built
error: build of '/nix/store/vlmf0pkf8fshclazwcl9h7nqy3faw5aa-review-shell.drv' failed

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/131808

15 packages failed to build:
ameba crystal crystal2nix crystal_0_33 crystal_0_34 crystal_0_35 crystal_0_36 icr lucky-cli mint oq scry shards shards_0_11 thicket

warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
error: build log of '/nix/store/ip70p46b8qa26y672xrj941b6ls18iiw-ameba-0.14.3.drv' is not available
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
error: build log of '/nix/store/2wsq6n63cnwhzanwrb5f97y9wybpmg3c-ameba-0.14.3' is not available
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
error: build log of '/nix/store/mdrw66hv4n4qfr6ahq73ygjxlyqf4lg4-crystal-1.0.0.drv' is not available
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
error: build log of '/nix/store/rjb9bpvr2lcw7m87rgvchkik8hvvp3ql-crystal-1.0.0' is not available
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
error: build log of '/nix/store/r31j1lk7zbf40bv056v9rjqz1z579wka-crystal2nix-0.1.1.drv' is not available
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
error: build log of '/nix/store/d0pflah4aa63chfr88k3iazs995msrlp-crystal2nix-0.1.1' is not available
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
$ nix-shell /Users/logan/.cache/nixpkgs-review/pr-131808/shell.nix

[nix-shell:~/.cache/nixpkgs-review/pr-131808]$ crystal
bash: crystal: command not found
```

`-f` didn't work for me, so the spew below is a deliberate omission of the flag. I didn't see the flag in the help as well. That said there's fresh git activity, and the outcome changed so I think we're okay. In any case the entire log is included below.

The skinny is that `crystal_0_34` failed (which causes 35 and 36 to also fail). `crystal_1_0` succeeds! I'm happy to continue testing other versions so we can determine exactly what does and does not build on Darwin.

``` sh
@ nixpkgs-review pr 131808   
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/131808/head:refs/nixpkgs-review/1
remote: Enumerating objects: 1650, done.
remote: Counting objects: 100% (1195/1195), done.
remote: Compressing objects: 100% (351/351), done.
remote: Total 1650 (delta 932), reused 1041 (delta 820), pack-reused 455
Receiving objects: 100% (1650/1650), 1.75 MiB | 5.19 MiB/s, done.
Resolving deltas: 100% (1032/1032), completed with 295 local objects.
From https://github.com/NixOS/nixpkgs
   3cc3160fe6c..9978f777fce  master                -> refs/nixpkgs-review/0
   2d9674ecf56..a2c0d744f84  refs/pull/131808/head -> refs/nixpkgs-review/1
$ git worktree add /Users/logan/.cache/nixpkgs-review/pr-131808-1/nixpkgs 9978f777fce61f6a053027d54388caed7ebb7a42
Preparing worktree (detached HEAD 9978f777fce)
Updating files: 100% (27208/27208), done.
HEAD is now at 9978f777fce Merge pull request #131954 from prusnak/python-trezor
$ nix-env --option system x86_64-darwin -f /Users/logan/.cache/nixpkgs-review/pr-131808-1/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit --no-ff a2c0d744f84d2e5a0e21f5d1717e645b82c2670b
Automatic merge went well; stopped before committing as requested
$ nix-env --option system x86_64-darwin -f /Users/logan/.cache/nixpkgs-review/pr-131808-1/nixpkgs -qaP --xml --out-path --show-trace --meta
16 packages updated:
ameba crystal crystal crystal crystal crystal crystal2nix icr-unstable lucky-cli mint oq scry shards shards shards thicket

warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
warning: ignoring the user-specified setting 'system', because it is a restricted setting and you are not a trusted user
$ nix --experimental-features nix-command build --no-link --keep-going -f /Users/logan/.cache/nixpkgs-review/pr-131808-1/build.nix
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
copying path '/nix/store/715y5lnw12vdcc3hh9sc9vbkdi6c83x6-bash-4.4-p23' from 'https://cache.nixos.org'...
copying path '/nix/store/4npah0kdy1jgqsbb4sw2vpbv9l2zsh18-Libsystem-1238.60.2' from 'https://cache.nixos.org'...
copying path '/nix/store/ckbmjdmsa2gx240sz4vr9z8hivn9gp5x-Security-55471.14.18' from 'https://cache.nixos.org'...
copying path '/nix/store/vbmi4ha8c014q3bri9iriggbd8hqdqss-adv_cmds-119-locale' from 'https://cache.nixos.org'...
copying path '/nix/store/00ngdc8ay9srb7ckm99kx6ims2siwqgq-apple-framework-CFNetwork' from 'https://cache.nixos.org'...
copying path '/nix/store/87y9iidffga2j5grsdvvwfj76l6xr6lb-apple-framework-CoreBluetooth' from 'https://cache.nixos.org'...
copying path '/nix/store/chpxcjxdq0cyxcs5dc2zkkaxivwih0jj-apple-framework-CoreData' from 'https://cache.nixos.org'...
copying path '/nix/store/bx78qsqny0bfaf6s6idjmjx83j470n3r-apple-framework-CoreFoundation' from 'https://cache.nixos.org'...
copying path '/nix/store/68yn2imvrrw86mbiwkk195n16fmm2f7f-apple-framework-CoreImage' from 'https://cache.nixos.org'...
copying path '/nix/store/3zynnka2fj57s030kdvp5121cb5mrsvf-apple-framework-CoreMIDI' from 'https://cache.nixos.org'...
copying path '/nix/store/5hpyrir2f419dch43nlkz0mjzznnyw1j-apple-framework-IOKit' from 'https://cache.nixos.org'...
copying path '/nix/store/3wzawr2vj64y2l47nak2vvwckzci8d64-apple-framework-Metal' from 'https://cache.nixos.org'...
copying path '/nix/store/p64d7siwnvf3fpscjzifgn90q859wn3w-apple-framework-CoreAudio' from 'https://cache.nixos.org'...
copying path '/nix/store/2m4j2za4hmj83f0677yiznff129vdf3k-apple-framework-DiskArbitration' from 'https://cache.nixos.org'...
copying path '/nix/store/yjamjr77klbn539xib8vamx5sp6wcgzs-apple-framework-AudioToolbox' from 'https://cache.nixos.org'...
copying path '/nix/store/bisagmfn1vpfswv1r1vpa1d8hh90scsz-apple-framework-IOBluetooth' from 'https://cache.nixos.org'...
copying path '/nix/store/lpzfxmhf3z77qgpcxc3zfrbb2czzgx4q-apple-framework-NetFS' from 'https://cache.nixos.org'...
copying path '/nix/store/mzzra6k9pi86yasa7x7ql4rc2mdfapi1-apple-framework-OpenDirectory' from 'https://cache.nixos.org'...
copying path '/nix/store/c072idq716ibax9jv8rhyb64rhnxzmfj-apple-framework-OpenGL' from 'https://cache.nixos.org'...
copying path '/nix/store/r1m56hdkhap9pcnngp5lzmgwa4yl1x29-apple-framework-Security' from 'https://cache.nixos.org'...
copying path '/nix/store/z3yld7y61yiklysjzf6k6kvfihyscxgb-apple-framework-SecurityFoundation' from 'https://cache.nixos.org'...
copying path '/nix/store/i6rzv6vw3zx060w6r4spbvlgpdjgg2k4-apple-framework-ServiceManagement' from 'https://cache.nixos.org'...
copying path '/nix/store/b16jnrlaclssglx8zszp4dwp0i893y5w-apple-framework-CoreWLAN' from 'https://cache.nixos.org'...
copying path '/nix/store/dh67n9z3mhlmmcj4nyk7pm9nhcas9s8y-apple-framework-CoreServices' from 'https://cache.nixos.org'...
copying path '/nix/store/645ys9v8wd3grfvln7mgvnc8854yhdrl-apple-framework-Accelerate' from 'https://cache.nixos.org'...
copying path '/nix/store/crz75rnr1xsp57snim91irbrgi2kxi5d-apple-framework-SystemConfiguration' from 'https://cache.nixos.org'...
copying path '/nix/store/ncyf8lxiaj4j3d1c3fwxkghy2vjwbshl-apple-framework-UIFoundation' from 'https://cache.nixos.org'...
copying path '/nix/store/yhc9gnqw4vm3b84xqcyfxalph5bng2cg-apple-lib-xpc' from 'https://cache.nixos.org'...
copying path '/nix/store/fx1yy6wzxm2w6b6xv177ryraa7vzjqa1-brotli-1.0.9-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/bzkf6rs464xs44xqkcrkb0czidb122jg-apple-framework-IOSurface' from 'https://cache.nixos.org'...
copying path '/nix/store/5f1qxwxxp9mag9r91mdfs3w1x3y99kxm-bzip2-1.0.6.0.2' from 'https://cache.nixos.org'...
copying path '/nix/store/2qnpghhvim4yv0cb0ds5swcll09k841m-apple-framework-CoreGraphics' from 'https://cache.nixos.org'...
copying path '/nix/store/a2y06jcg1jsghnrakw4qr27vlm1cdzym-apple-framework-OpenCL' from 'https://cache.nixos.org'...
copying path '/nix/store/kdnlp3kz0x7wbagn960i3rw657ws27hq-apple-framework-CoreText' from 'https://cache.nixos.org'...
copying path '/nix/store/y70lq54ix9qc0hcahnr452z1hag6w5sq-apple-framework-ImageIO' from 'https://cache.nixos.org'...
copying path '/nix/store/j998frnsa1hxkfbysq0bkw6hjfrd928v-bzip2-1.0.6.0.2-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/7zjv0kdb724q9lpjgx9k5wk58dg1h47v-apple-framework-ApplicationServices' from 'https://cache.nixos.org'...
copying path '/nix/store/fqflvsqh0hkmagb6ci3r775h3m30q7hs-curl-7.76.1-man' from 'https://cache.nixos.org'...
copying path '/nix/store/1h653vqxkd69ylzgxbdrj4sx7idki7g5-apple-framework-CoreVideo' from 'https://cache.nixos.org'...
copying path '/nix/store/7i321ai77n1y10vdvb7rsnk1ng8vp11b-ed-1.17' from 'https://cache.nixos.org'...
copying path '/nix/store/8jv49b53ybn6fnwhd0dm2vk42w9zzpcv-expand-response-params' from 'https://cache.nixos.org'...
copying path '/nix/store/phq0s0i1vhlvs7y77gyygnyd1zm76mlh-gawk-5.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/k5szspjyr3axcj91vzmdy0imyswyc7fa-git-2.32.0-doc' from 'https://cache.nixos.org'...
copying path '/nix/store/aqrq990va9j1v6gvrgxksr5av9vpcxz3-gnumake-4.3' from 'https://cache.nixos.org'...
copying path '/nix/store/z6ih791dzyjvdqm8gl5glz065kk8zh7p-gnused-4.8' from 'https://cache.nixos.org'...
copying path '/nix/store/3n9fvf4ybdwizym6cwnjf6f57p7mvrab-gzip-1.10' from 'https://cache.nixos.org'...
copying path '/nix/store/gsn0h0inq6ybv0pbzzq1yv02gs2kpqnn-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/s7bqhhx7pwwif7rn708gnqq0giia5v89-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/61qlnnsv119vbwww6800j19y4d3lgkg1-install-shell-files' from 'https://cache.nixos.org'...
copying path '/nix/store/v88kadzxxwdkwmksjprhabjpdz43ic49-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/kjnik2iwgnc11ylm2msz7kgqz5wjyyir-libatomic_ops-7.6.10' from 'https://cache.nixos.org'...
copying path '/nix/store/0yrrpxkc6l1kvdf491qa1m7scamsmqb7-libcxxabi-7.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/6jzg0m5y8pqi33dkg8xfglwvjym8xj35-libatomic_ops-7.6.10-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/fsz5rxl37pamwhqkmp9w4jryfc4pfky3-libcxx-7.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/y736kjvzdclbzcxdv5fc9ghb832fq0i7-libcxxabi-7.1.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/r2w4s36nr4w9gnchz4nyzj64nyaig9cr-ICU-66108' from 'https://cache.nixos.org'...
copying path '/nix/store/55nwvpwcx5hvp9ky2yf3n38xfj171idl-compiler-rt-libc-7.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/wkgrylws6qa3iajc2py42m5fhyxnwkjv-gmp-6.2.1' from 'https://cache.nixos.org'...
copying path '/nix/store/lw8gcf5givnwpflpgfkmayfq6by40wfr-compiler-rt-libc-7.1.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/crjmgrws5w2pjhm5k887m93rrkcf36hb-coreutils-8.32' from 'https://cache.nixos.org'...
copying path '/nix/store/znv06z6pld0ivwddj52rishggwc2m94c-libcxx-7.1.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/adigdkh0pibb4afqf5q2sv3503ln0iyb-diffutils-3.7' from 'https://cache.nixos.org'...
copying path '/nix/store/2fzds9cqx148i1lf84sc9i0982bv9rbx-findutils-4.8.0' from 'https://cache.nixos.org'...
copying path '/nix/store/zyh3yfv4npb5vyhqmp6hswfps0j7smyn-libffi-3.3' from 'https://cache.nixos.org'...
copying path '/nix/store/qgygs5g9l0zxhs8fzjjqpmmlzps1hlrd-libiconv-50' from 'https://cache.nixos.org'...
copying path '/nix/store/5vw0211970xi853z6l9a42s0kdbzv3b0-libkrb5-1.18' from 'https://cache.nixos.org'...
copying path '/nix/store/ig8x9si4x7kwxv2qimvdwk2wpafv21gp-gettext-0.21' from 'https://cache.nixos.org'...
copying path '/nix/store/wmanif8d0cqg7vwzd2ps3ix6gvq57764-mirrors-list' from 'https://cache.nixos.org'...
copying path '/nix/store/q7wxq69v35d0bxiqrycldz00wg9kxqzd-gnutar-1.34' from 'https://cache.nixos.org'...
copying path '/nix/store/fci25bpkavmg481ws49qdjvv9k4vmm6y-ncurses-6.2' from 'https://cache.nixos.org'...
copying path '/nix/store/xd59il5rf35cknc5i9ckf50y1dp8zjh4-ncurses-6.2-man' from 'https://cache.nixos.org'...
copying path '/nix/store/dw0qfmxss8if5ks8qr0rk2sb8wnng3dw-libtapi-1100.0.11' from 'https://cache.nixos.org'...
copying path '/nix/store/47svf9r3ww50svr3f5nkav7bprvhbwg0-ncurses-6.2-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/6biazmfb7jq7y0zxqf4r6b3nxx24n8f9-cctools-port-949.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/fkaxqpng0z2dmcqq7wm9d8zsv20r8l4n-nghttp2-1.43.0' from 'https://cache.nixos.org'...
copying path '/nix/store/bk8h2xqxsxrvbdf96cgdr5xvw2yafp8j-nghttp2-1.43.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/a8p7ndykav8k7a34n2q1asxkfznshpfg-nix-2.3.14-man' from 'https://cache.nixos.org'...
copying path '/nix/store/n4g6bc1gs1jjhx1hzznqh3950ix60v5i-objc4-709.1' from 'https://cache.nixos.org'...
copying path '/nix/store/rsza0nf7wwmv0c17q0x9yhf79ycvqgm1-openssl-1.1.1k' from 'https://cache.nixos.org'...
copying path '/nix/store/gswa3ywvzhm145g8jjhaihsa8qyjmc06-apple-framework-Foundation' from 'https://cache.nixos.org'...
copying path '/nix/store/3yrzfhqvq41nh8fi8mjm77hwhaqprzff-apple-framework-QuartzCore' from 'https://cache.nixos.org'...
copying path '/nix/store/cqadarnj29fpv1vm8bzbh6dpb8aaxzjz-patch-2.7.6' from 'https://cache.nixos.org'...
copying path '/nix/store/jrbfjc704hxs70q3ljjpyl0f4j7a57k9-apple-framework-Carbon' from 'https://cache.nixos.org'...
copying path '/nix/store/mg2ar6f1r2nssalsss32lyqifwlhhmzb-pcre-8.44' from 'https://cache.nixos.org'...
copying path '/nix/store/jiwzfi73h9zadmnb6qgjlwjisan3sznd-apple-framework-AudioUnit' from 'https://cache.nixos.org'...
copying path '/nix/store/lr3fvwhvd8zpmjbibd3h02qdx85mxrv4-gnugrep-3.6' from 'https://cache.nixos.org'...
copying path '/nix/store/6i6c9gz5d79b97gnr0a12yn3d49dyrqy-apple-framework-AppKit' from 'https://cache.nixos.org'...
copying path '/nix/store/j9d2ic7lbrfpai3iyn0m55rxagrjh2j1-pcre-8.44-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/wpx3d215y08c37sc5fzlic5hkn8hhha4-apple-framework-Cocoa' from 'https://cache.nixos.org'...
copying path '/nix/store/c7pd7p0y85vifzvkv1rfrz9dxqykg4xx-pcre-8.44-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/cz81sbpnda675pfq7gfnsdr9k8llgfw4-hidapi-0.10.1' from 'https://cache.nixos.org'...
copying path '/nix/store/dlh18gcgvy32811cnirjmadpn9x6lahg-perl5.34.0-Encode-Locale-1.05' from 'https://cache.nixos.org'...
copying path '/nix/store/mwhc7d6g1zzrvjvqwkf988gbf5q03lrf-perl5.34.0-FCGI-ProcManager-0.28' from 'https://cache.nixos.org'...
copying path '/nix/store/cklqcidnwvvmp6h9gpdbnldmkak0mz4h-perl5.34.0-HTML-TagCloud-0.38' from 'https://cache.nixos.org'...
copying path '/nix/store/b73lrrsjx4bvp8qh7jairv55syvagkwd-perl5.34.0-HTML-Tagset-3.20' from 'https://cache.nixos.org'...
copying path '/nix/store/5kkpswp4zyzwwfwhccaszvrrfwifxg6r-perl5.34.0-IO-HTML-1.004' from 'https://cache.nixos.org'...
copying path '/nix/store/p7pw00idxa2b4ywb207d29qik09bs5h3-perl5.34.0-LWP-MediaTypes-6.04' from 'https://cache.nixos.org'...
copying path '/nix/store/156rmc1d7m0w0awhdn8jky03i1nyf7rh-perl5.34.0-Test-Needs-0.002006' from 'https://cache.nixos.org'...
copying path '/nix/store/q647daz7i71y30birkp6gls1r6fffwp2-perl5.34.0-Test-RequiresInternet-0.05' from 'https://cache.nixos.org'...
copying path '/nix/store/qc7hpsj9h68fnidcbvq4g6lih3666wv7-perl5.34.0-TimeDate-2.33' from 'https://cache.nixos.org'...
copying path '/nix/store/4ij9w5ck9m2ijpqpfjz46cnsa01nlibl-perl5.34.0-Try-Tiny-0.30' from 'https://cache.nixos.org'...
copying path '/nix/store/clgjx9nnbgcb5zgndhn45b2kas7d3ily-perl5.34.0-HTTP-Date-6.05' from 'https://cache.nixos.org'...
copying path '/nix/store/6icby0hr0lp7gswwqq207b8lvvjxi9wq-perl5.34.0-Test-Fatal-0.016' from 'https://cache.nixos.org'...
copying path '/nix/store/h1cjrnwdmvmrh2mspkrr0agv0qn257dj-perl5.34.0-File-Listing-6.14' from 'https://cache.nixos.org'...
copying path '/nix/store/xcn2cfrpql8panqjqhfxd4sqlzh5ja66-perl5.34.0-URI-5.05' from 'https://cache.nixos.org'...
copying path '/nix/store/a74jwf4lgca5d05klna9slxydgdsngk9-tzdata-2021a' from 'https://cache.nixos.org'...
copying path '/nix/store/mbm45kwp1vqwnk52cp0w9dy4f0qsxljb-perl5.34.0-HTTP-Message-6.26' from 'https://cache.nixos.org'...
copying path '/nix/store/9wy7j2x17zbg471ag8n8wnvakf2dga4k-perl5.34.0-Net-HTTP-6.19' from 'https://cache.nixos.org'...
copying path '/nix/store/njfdj5rp1r3yc1sv9r9ffph945928nx6-perl5.34.0-HTTP-Cookies-6.09' from 'https://cache.nixos.org'...
copying path '/nix/store/kmmqfcwps62k79yb5rljgkydff5mkgfp-perl5.34.0-HTTP-Daemon-6.01' from 'https://cache.nixos.org'...
copying path '/nix/store/x9chz553wdb3j2hpq2z5pgg43ysrq87n-perl5.34.0-HTTP-Negotiate-6.01' from 'https://cache.nixos.org'...
copying path '/nix/store/baa2clyal7jbzjfcaj62sjzjpm407ixh-perl5.34.0-WWW-RobotRules-6.02' from 'https://cache.nixos.org'...
copying path '/nix/store/mq766n41v9djm2rf74dpg4hd7n5mbwwh-xz-5.2.5' from 'https://cache.nixos.org'...
copying path '/nix/store/x87bqc42xmskjwf5c2a9bbzkd8aqwrca-zlib-1.2.11' from 'https://cache.nixos.org'...
copying path '/nix/store/00vcq1i9x1dwwn21dqkagybcxa1lcnwi-xz-5.2.5-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/wia4ggzr91yglpq6z1lg48qnv5z4yy94-binutils-2.35.1' from 'https://cache.nixos.org'...
copying path '/nix/store/ggr82sipdfkxszl9qnbd0zg0hlqr7b1c-libssh2-1.9.0' from 'https://cache.nixos.org'...
copying path '/nix/store/0idd3d3y4vbiilwzhjq1my4npac6bp3r-libxml2-2.9.12' from 'https://cache.nixos.org'...
copying path '/nix/store/niqwc81ynr8csa9rdz9l0zg2qcjhzspb-curl-7.76.1' from 'https://cache.nixos.org'...
copying path '/nix/store/fbhp0ph0366r5614xlincnvx6qj7rlsi-llvm-7.1.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/w9vgngyar0ykvmamm6k0bf5k9yn9kbhb-swift-corefoundation' from 'https://cache.nixos.org'...
copying path '/nix/store/xhijdap7ln3nw867cq6m3knshww9f2rn-clang-7.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/gfnka54yxsy9nlnyy5wkyh60i7ax43lg-aws-c-common-0.6.8' from 'https://cache.nixos.org'...
copying path '/nix/store/sbggvnipiy6x3g7xpmfgjybajnlfyl5x-boehm-gc-8.0.4' from 'https://cache.nixos.org'...
copying path '/nix/store/y1ivad5rmn73vrrvn8v3x55xfcc32kbv-aws-c-cal-0.5.11' from 'https://cache.nixos.org'...
copying path '/nix/store/3sq2cn4b0n94hxdx8yk2h38l1vma38w2-aws-checksums-0.1.11' from 'https://cache.nixos.org'...
copying path '/nix/store/k28jza7gvv5h18nn1lh0zawzga363bg8-aws-c-io-0.9.1' from 'https://cache.nixos.org'...
copying path '/nix/store/xcq8mi7rinc85m0wpzr5az617ig2vs8s-boehm-gc-8.0.4' from 'https://cache.nixos.org'...
copying path '/nix/store/n21xj8xnv420k9ypixb4ypf1a8fcd2h1-aws-c-event-stream-0.2.7' from 'https://cache.nixos.org'...
copying path '/nix/store/kj2vz3p10bl45fl9pglsnq4xf9sihp3d-boehm-gc-8.0.4-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/f2qq98ksl340z3ncwcwx6fcjp4mc6iyy-brotli-1.0.9-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/69iqwg8ksw89fhzk8x2n7lj98hfj3lmj-c-ares-1.17.1' from 'https://cache.nixos.org'...
copying path '/nix/store/aa3xbipcdryaf3mn9ic5y1smzlxngvfz-clang-7.1.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/3zl3bz87y2f06cjmknasp44dxq049400-compiler-rt-libc-10.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/djk25cq1wr1vs7qbybzak4a6alhrsd8x-configd-453.19' from 'https://cache.nixos.org'...
copying path '/nix/store/3zvx384vs3fid577ny8h9n3j0w4vgj4r-compiler-rt-libc-10.0.1-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/jzcax18k0yxhg1jxrx7j421n47j3dllc-editline-1.17.1' from 'https://cache.nixos.org'...
copying path '/nix/store/m9q3wwchxn9938hri6208j7cim697ah5-expand-response-params' from 'https://cache.nixos.org'...
copying path '/nix/store/7a27b4f7c3sqbjc1vg0fjnplkqd2b27d-expat-2.4.1' from 'https://cache.nixos.org'...
copying path '/nix/store/zk81d86gikl6dm96kfylgd2yi7ji8qbm-gdbm-1.20' from 'https://cache.nixos.org'...
copying path '/nix/store/hdrwrac2r06nmcifk2asg2j7hinb5qpb-gettext-0.21' from 'https://cache.nixos.org'...
copying path '/nix/store/b351b0jfgalgniw5pazsjvrf3jzjgfsm-icu4c-69.1' from 'https://cache.nixos.org'...
copying path '/nix/store/jggqvhq5db9c0q635i8r8qvlc30g72sv-libcbor-0.8.0' from 'https://cache.nixos.org'...
copying path '/nix/store/69m1630x5zhf9ss7p41dhz1av1bjsskj-boost-1.69.0' from 'https://cache.nixos.org'...
copying path '/nix/store/pq7jlnvhz522kjyaqmhzv4g0yiwkscyf-libcxx-10.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/8aaj88rlk3d88znxw29sh9nhmqlf4kcw-libcxxabi-10.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/rx4zyp64rr8jcls6pxmmy31n6gdawxx3-libcxx-10.0.1-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/h18pm9ddgx2aj2y1jw8gcgxvwhj4bvbh-libcxxabi-10.0.1-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/nvypxmbcfc8yxrpgsgxhbnp6s2d4a5az-libedit-20210522-3.1' from 'https://cache.nixos.org'...
copying path '/nix/store/w4fi1l49kl3hri8hr8advfxkz8ydibfb-libev-4.33' from 'https://cache.nixos.org'...
copying path '/nix/store/n23gm30ab7nf3qcfkq13019s8nx1p8f1-libevent-2.1.12' from 'https://cache.nixos.org'...
copying path '/nix/store/mrdsjbp54ys55dmav5501k59jqk5vy2b-libkrb5-1.18' from 'https://cache.nixos.org'...
copying path '/nix/store/kl9m508wkx7q4ysapa2zbvj8nj2x72hc-libsodium-1.0.18' from 'https://cache.nixos.org'...
copying path '/nix/store/4xp86qi542rlddjxndff141l4cyqw9ri-libxml2-2.9.12' from 'https://cache.nixos.org'...
copying path '/nix/store/b2h7sv70fdma61nby6k601nalv88ww27-libyaml-0.2.5' from 'https://cache.nixos.org'...
copying path '/nix/store/mja1sc7w9iy854xfycl0l669zgdlr88r-libxml2-2.9.12-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/ka0ibmgpdrn1xqq88dad4g4bvp0s1gim-llvm-10.0.1-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/np1cc9lc9jn9pxbfzkgbz24c8pxbibwa-llvm-7.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/nfkxc197zb3an9zd4scxdg5i6jkyq8rf-clang-10.0.1-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/falx4gakl6k1q64aqr4f8mvy8vfqcqaq-cctools-binutils-darwin-949.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/n3zqmi8n898q09n6ji5a9pi3860x43k7-clang-10.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/4j3q5kmmdwkgppn6g8hna8wg6a4zrvxd-cctools-binutils-darwin-wrapper-949.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/g46mpzrx4qk0dzwnkkb024bkwzkscqsm-llvm-10.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/h5f6wzb72s0ihdsi2idvr622b6i3lfxi-clang-wrapper-10.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/jx31ah14p6pf488i7wwd88y2izmxi0wj-clang-wrapper-7.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/1ahb54by7x5h7yn8703nq25fiwdzg4ix-nghttp2-1.43.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/y240fpxd146jyl5760bbcr095sl8lyal-nghttp2-1.43.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/mf2i5vfp4dn657ch8q4rchpv2figd20d-onig-6.9.6' from 'https://cache.nixos.org'...
copying path '/nix/store/shp60sgvnh5is0mcd7w86gq531ws3sxn-openssl-1.1.1k' from 'https://cache.nixos.org'...
copying path '/nix/store/v4yia5ijgd42ld2dms4jzivyvi0xh1g8-jq-1.6-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/iqv2zgw2x6h4b3w5m6xy7a481xy7n7dh-libevent-2.1.12-openssl' from 'https://cache.nixos.org'...
copying path '/nix/store/hmmba0bqn54xw13w8hm2rcdy504lr9js-jq-1.6-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/nmd2030g18847z6q0jchhh4cm57nj9k5-libevent-2.1.12-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/n8sa938iyszwj7508xcp3qhrp7pni2z2-jq-1.6-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/bmxvqnmx5afyvhk3iciapgfd71yxdzp7-libfido2-1.7.0' from 'https://cache.nixos.org'...
copying path '/nix/store/m8jmp0ffs11r0bigxd107v4j58r1amss-libssh2-1.9.0' from 'https://cache.nixos.org'...
copying path '/nix/store/pzjrm7c57jqpsajpbfzrx093l7dm60v6-openssh-8.6p1' from 'https://cache.nixos.org'...
copying path '/nix/store/b2wj786d4qizfamzwhc6qfsb1yaaqsm5-curl-7.76.1' from 'https://cache.nixos.org'...
copying path '/nix/store/vx0qk64p1rxf7px1gh1dpvxj4vq68zkl-openssl-1.1.1k' from 'https://cache.nixos.org'...
copying path '/nix/store/449hccl05dl6izr1w58fkfvgg5gzaaf2-aws-sdk-cpp-1.8.121' from 'https://cache.nixos.org'...
copying path '/nix/store/8gmbh9052963v5j9hr2j0lcw8vi67n2p-libssh2-1.9.0' from 'https://cache.nixos.org'...
copying path '/nix/store/bnyf85f6ywvr9ddvczc9wkwkggrm7saa-nghttp2-1.43.0-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/v0g00as1ca4n4z7b4757f5s8c0940d1x-curl-7.76.1' from 'https://cache.nixos.org'...
copying path '/nix/store/3qzxmqyxapyb9lidnm4mwl9wzxlywcdf-libssh2-1.9.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/1hm4pbl2an5zapv9wlgwqc9qhvynq3jv-curl-7.76.1-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/fbavqwhx3y3ljqa160a54cz0d4kgfjy6-nghttp2-1.43.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/mjlkzmwrih5bz2hzi9yzm5ngridf0n53-pcre2-10.36' from 'https://cache.nixos.org'...
copying path '/nix/store/1xjwb0bd02n1y621mjqani68kh6n6gig-perl-5.34.0' from 'https://cache.nixos.org'...
copying path '/nix/store/6wf3aa69wzi83lxz1gy41y90fiypz767-perl5.34.0-FCGI-0.79' from 'https://cache.nixos.org'...
copying path '/nix/store/gh4bqx29xyharyzbiqhdkd5c53xnx6vj-openssl-1.1.1k-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/prficcpfq5gs9353x6pnw9j32cc6k62s-openssl-1.1.1k-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/v6wm134pxiy4c0gz8l4p5h6fr6m00x2z-openssl-1.1.1k-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/2w0hc4n7szqicn8z4bcignxchj5xj0f2-openssl-1.1.1k-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/sdbsgmdc9xvlngvd2m2nr2vvzyzbcmyh-perl5.34.0-HTML-Parser-3.75' from 'https://cache.nixos.org'...
copying path '/nix/store/xz9n2vz4jpkq81x0wxjsxg986gc4pjz1-perl5.34.0-TermReadKey-2.38' from 'https://cache.nixos.org'...
copying path '/nix/store/8a8c4ahdnnp3p8nf1r4q2g881yz0cp7v-perl5.34.0-CGI-4.51' from 'https://cache.nixos.org'...
copying path '/nix/store/qmzjj3g1rbqyhi84ii8chr0g7sh8ahm0-perl5.34.0-libwww-perl-6.49' from 'https://cache.nixos.org'...
copying path '/nix/store/5mkv08mg9kk2gx12j0abfp5x348yf691-perl5.34.0-CGI-Fast-2.15' from 'https://cache.nixos.org'...
copying path '/nix/store/f6kpgxxhxvwdmhni33n3hj64hxsbwfww-pkg-config-0.29.2' from 'https://cache.nixos.org'...
copying path '/nix/store/mkr49brmqqad1yjac4lgffl1rjk0bf6m-readline-6.3p08' from 'https://cache.nixos.org'...
copying path '/nix/store/krhkk38bd42357y2z158l3r09k2mdj51-pkg-config-wrapper-0.29.2' from 'https://cache.nixos.org'...
copying path '/nix/store/z53fv57fgv1glng52lrdmfk482n6sb4i-readline-6.3p08-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/w7g29cf0s3g24kzxjf3k7brf2zr2djlc-shell_cmds-187' from 'https://cache.nixos.org'...
copying path '/nix/store/va4ipz29l8a1v9gvw7x3r5pp5zwv254b-sqlite-3.35.5' from 'https://cache.nixos.org'...
copying path '/nix/store/14357agl0nszd1ipy4aslbdan6sh6205-hostname-shell_cmds-187' from 'https://cache.nixos.org'...
copying path '/nix/store/0dalflwmfv7h2s077dgb3yn9lr3zv8a2-nix-2.3.14' from 'https://cache.nixos.org'...
copying path '/nix/store/ad19ya4siaxhz63w0m8xh97067iahks0-python3-3.9.6' from 'https://cache.nixos.org'...
copying path '/nix/store/3yq0m0kprzf172ljfwdl6sidvf6v6j9i-stdenv-darwin' from 'https://cache.nixos.org'...
copying path '/nix/store/ihlm88458pygspjfd44rzfl304rg362s-git-2.32.0' from 'https://cache.nixos.org'...
copying path '/nix/store/j7yi8i02lw1alasbsy13k3q0cdhhi7rr-stdenv-darwin' from 'https://cache.nixos.org'...
copying path '/nix/store/qpcpd7jqikzcn31wrnk1v57wh2pdr2yd-nix-prefetch-git' from 'https://cache.nixos.org'...
copying path '/nix/store/72cnljimi7v8nba6vwpibxhwvm4ymzqf-which-2.21' from 'https://cache.nixos.org'...
copying path '/nix/store/r9sksj7xmm0aiysm2s78m9irq4bvg6cb-zlib-1.2.11-dev' from 'https://cache.nixos.org'...
building '/nix/store/0dbd5m06n05rghk3m43825grz6bq7rh3-crystal-lib.drv'...
copying path '/nix/store/bbyrdfcb7psnxs8vw8smzawgdsfxl0m4-curl-7.76.1-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/b2gy9v88nx2lnznbby4if1jsv86schry-libxml2-2.9.12-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/33dwgivfxky71rrpwaibfd18y92mzgqf-llvm-10.0.1-dev' from 'https://cache.nixos.org'...
building '/nix/store/k20ifidlxzqkn2lwy2ljb4bmgbl2w9an-crystal-0.34.0-1-darwin-x86_64.tar.gz.drv'...
building '/nix/store/03ngcrarpa7f99h7dj7f3bh4nk4z1gyl-crystal-1.0.0-1-darwin-x86_64.tar.gz.drv'...
building '/nix/store/avq550j5h0hhk3jzclxhsyhlnacr70yr-crystal-binary-0.34.0.drv'...
building '/nix/store/j8w64spv0mks2pi9qg1s9a2xxk4anax0-crystal-binary-1.0.0.drv'...
building '/nix/store/ifpfcykk5ksl0fkfrrvnjrkssl4qng82-crystal-0.34.0.drv'...
builder for '/nix/store/ifpfcykk5ksl0fkfrrvnjrkssl4qng82-crystal-0.34.0.drv' failed with exit code 2; last 10 log lines:
    Semantic: instance var doesn't infer type to be nilable if using self.class in call in assign
    Semantic: pointer allows using pointer with subclass
    Semantic: primitives types pointer of int
  
  Failures:
  
    1) Crystal::Command::FormatCommand formats stdin (bug + show-backtrace)
  
  Error running at_exit handler: Nil assertion failed
  make: *** [Makefile:89: compiler_spec] Error 1
cannot build derivation '/nix/store/c11kzwnkbgd3mhxarjkn0mqbv8n72c6w-crystal-0.35.1.drv': 1 dependencies couldn't be built
building '/nix/store/z39jzs86bw3yfrmvf3nls7zlbvfmvnnd-crystal-1.0.0.drv'...
cannot build derivation '/nix/store/qmwscb3xyla6awyjhnc7jhyb8638s6aw-shards-0.11.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/fvgz7mfad0yry8kkvr8kmpay85s5rjxq-crystal-0.36.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/gyzgfndrnw78jbixm4ahr35aq1fdk46m-shards-0.14.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/nw1qqfgj2q9sq6naa329lgmpz03wjy25-ameba-0.14.3.drv': 1 dependencies couldn't be built
building '/nix/store/bvzn200ng95563w6n50gicindmygk357-crystal-lib.drv'...
cannot build derivation '/nix/store/91d381prjwss14y9709j27x3a0lsqywx-oq-1.2.0.drv': 1 dependencies couldn't be built
building '/nix/store/fcmji6azmn7wchjbzlwg517j9171klza-crystal-lib.drv'...
cannot build derivation '/nix/store/7axrr9nkr5f6wh3yynj5qmd1cd4mrzx4-icr-unstable-2021-03-14.drv': 1 dependencies couldn't be built
building '/nix/store/fqb1qsizwjpczhihfjsjmmlgl6bg1gp0-crystal-lib.drv'...
cannot build derivation '/nix/store/x1jp4z7jf8hprnhs7ipl8h16h6fb3c50-crystal2nix-0.1.1.drv': 1 dependencies couldn't be built
building '/nix/store/qqgv8rpgpp2lpx8q78qbg6573yn2zr2q-crystal-lib.drv'...
building '/nix/store/vmyg9hg718r6inaxb3sjmryb10sn9n0h-crystal-lib.drv'...
cannot build derivation '/nix/store/47s7bs1w64lfssknvcj59iam0pa2mlwk-scry-0.9.1.drv': 1 dependencies couldn't be built
building '/nix/store/198gmd8cb8a63qjg8s11gid5wmhx8ank-lucky-cli-0.26.0.drv'...
cannot build derivation '/nix/store/3ahbm2nj8mw2ahhamgv3j3rh8fmjrjln-mint-0.14.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/wk9ldl5mna4lmcikddxvrab0g7xzbzd3-review-shell.drv': 11 dependencies couldn't be built
error: build of '/nix/store/wk9ldl5mna4lmcikddxvrab0g7xzbzd3-review-shell.drv' failed

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/131808

1 package marked as broken and skipped:
thicket

11 packages failed to build:
ameba crystal2nix crystal_0_34 crystal_0_35 crystal_0_36 icr mint oq scry shards shards_0_11

2 packages built:
crystal lucky-cli

warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
error: build log of '/nix/store/nw1qqfgj2q9sq6naa329lgmpz03wjy25-ameba-0.14.3.drv' is not available
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
error: build log of '/nix/store/2gfrz9plihxdnjxqsx43957p351cl2kg-ameba-0.14.3' is not available
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
$ nix-shell /Users/logan/.cache/nixpkgs-review/pr-131808-1/shell.nix
querying info about '/nix/store/jza3wx6kx3w7cbasjl78sm2w85qn9gi7-bash-interactive-4.4-p23-doc' on 'https://cache.nixos.org'...
querying info about '/nix/store/cn2gg00v5znz09qj7lpvwjkf0s1i1p1f-bash-interactive-4.4-p23-info' on 'https://cache.nixos.org'...
querying info about '/nix/store/kvfqqgipb2m5r6cjnwmq34jnvb92mqn4-bash-interactive-4.4-p23' on 'https://cache.nixos.org'...
querying info about '/nix/store/r9j2igrgwvpv8qnf23wmqh5dvgsbhd02-bash-interactive-4.4-p23-dev' on 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/jza3wx6kx3w7cbasjl78sm2w85qn9gi7.narinfo'...
downloading 'https://cache.nixos.org/kvfqqgipb2m5r6cjnwmq34jnvb92mqn4.narinfo'...
downloading 'https://cache.nixos.org/r9j2igrgwvpv8qnf23wmqh5dvgsbhd02.narinfo'...
downloading 'https://cache.nixos.org/cn2gg00v5znz09qj7lpvwjkf0s1i1p1f.narinfo'...
querying info about '/nix/store/svl0j5hxn6diywzgn0961kllvijzr00g-bash-interactive-4.4-p23-man' on 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/svl0j5hxn6diywzgn0961kllvijzr00g.narinfo'...
querying info about '/nix/store/vghj7xdadpckxqycfck0lq3jaji0cy40-readline-7.0p5' on 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/vghj7xdadpckxqycfck0lq3jaji0cy40.narinfo'...
these 6 paths will be fetched (1.46 MiB download, 7.85 MiB unpacked):
  /nix/store/cn2gg00v5znz09qj7lpvwjkf0s1i1p1f-bash-interactive-4.4-p23-info
  /nix/store/jza3wx6kx3w7cbasjl78sm2w85qn9gi7-bash-interactive-4.4-p23-doc
  /nix/store/kvfqqgipb2m5r6cjnwmq34jnvb92mqn4-bash-interactive-4.4-p23
  /nix/store/r9j2igrgwvpv8qnf23wmqh5dvgsbhd02-bash-interactive-4.4-p23-dev
  /nix/store/svl0j5hxn6diywzgn0961kllvijzr00g-bash-interactive-4.4-p23-man
  /nix/store/vghj7xdadpckxqycfck0lq3jaji0cy40-readline-7.0p5
copying path '/nix/store/jza3wx6kx3w7cbasjl78sm2w85qn9gi7-bash-interactive-4.4-p23-doc' from 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/nar/07n6wnnsa7c2vv42qbvgjc7brnwnsypgyc9lssvzl714cwn6nr1w.nar.xz'...
copying path '/nix/store/cn2gg00v5znz09qj7lpvwjkf0s1i1p1f-bash-interactive-4.4-p23-info' from 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/nar/13vmv4cy17vwwi31jc0yk4qmjjpvdgw7cxyyvk91wmwm8d8js05n.nar.xz'...
copying path '/nix/store/svl0j5hxn6diywzgn0961kllvijzr00g-bash-interactive-4.4-p23-man' from 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/nar/08di0250slfpvsfhm47fy4k6c5a745l0i4ywwjxkikbbq6j8yv9z.nar.xz'...
copying path '/nix/store/vghj7xdadpckxqycfck0lq3jaji0cy40-readline-7.0p5' from 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/nar/1k3s68mrhf8azm7p1a4ywa3991faj77xbk61n7iqhakaw9hqs0cd.nar.xz'...
copying path '/nix/store/kvfqqgipb2m5r6cjnwmq34jnvb92mqn4-bash-interactive-4.4-p23' from 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/nar/15qm2hha7w58n8xj2b4vs7klk3dwfqwx5pkynfq7gyzf4mn4n7q1.nar.xz'...
copying path '/nix/store/r9j2igrgwvpv8qnf23wmqh5dvgsbhd02-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/nar/04sw3wqr82kxszxiqvbzphhhcx9dafiw1n9w7r5k01sr7630fm0j.nar.xz'...

[nix-shell:~/.cache/nixpkgs-review/pr-131808-1]$ which crystal
/nix/store/rqbh8ir75i2i7d602hrs2zadpzrdcrgd-crystal-1.0.0/bin/crystal

[nix-shell:~/.cache/nixpkgs-review/pr-131808-1]$ crystal -v
Crystal 1.0.0 (2021-07-29)

LLVM: 10.0.1
Default target: x86_64-apple-macosx
```

I verified I'm on latest commit. Checkout call is included in the spew below.

Looks like 35 fails but 36 succeeds. 1.0 still looks good. I assume those other packages listed (`mint`, `oq`, `shards_0_11`) are not using 1.0 (yet). Of note, `shards` is working too :)

Thanks for iterating on this with me! Sorry it takes a while. My machine isn't doing much other than this and it takes a little almost 2 hours. I'm happy to test further though.

``` sh
$ gh pr checkout 131808 -f
remote: Enumerating objects: 7, done.
remote: Counting objects: 100% (7/7), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 7 (delta 0), reused 7 (delta 0), pack-reused 0
Unpacking objects: 100% (7/7), 11.18 KiB | 763.00 KiB/s, done.
From github.com:NixOS/nixpkgs
 * branch                    refs/pull/131808/head -> FETCH_HEAD
HEAD is now at 94cf85db5ac c

$ nixpkgs-review pr 131808          
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/131808/head:refs/nixpkgs-review/1
remote: Enumerating objects: 84, done.
remote: Counting objects: 100% (60/60), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 84 (delta 55), reused 58 (delta 55), pack-reused 24
Unpacking objects: 100% (84/84), 3.51 MiB | 3.92 MiB/s, done.
From https://github.com/NixOS/nixpkgs
   9978f777fce..3936313b1f1  master                -> refs/nixpkgs-review/0
   a2c0d744f84..94cf85db5ac  refs/pull/131808/head -> refs/nixpkgs-review/1
$ git worktree add /Users/logan/.cache/nixpkgs-review/pr-131808-2/nixpkgs 3936313b1f13c198ee1ae56009db55fd6a87ab02
Preparing worktree (detached HEAD 3936313b1f1)
Updating files: 100% (27208/27208), done.
HEAD is now at 3936313b1f1 nixos/firewall: document log location
$ git merge --no-commit --no-ff 94cf85db5acbc5b0ddbca116879dd43470fb4f56
Automatic merge went well; stopped before committing as requested
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
warning: ignoring the user-specified setting 'system', because it is a restricted setting and you are not a trusted user
$ nix --experimental-features nix-command build --no-link --keep-going -f /Users/logan/.cache/nixpkgs-review/pr-131808-2/build.nix
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
building '/nix/store/ifpfcykk5ksl0fkfrrvnjrkssl4qng82-crystal-0.34.0.drv'...
builder for '/nix/store/ifpfcykk5ksl0fkfrrvnjrkssl4qng82-crystal-0.34.0.drv' failed with exit code 2; last 10 log lines:
    Semantic: instance var doesn't infer type to be nilable if using self.class in call in assign
    Semantic: pointer allows using pointer with subclass
    Semantic: primitives types pointer of int
  
  Failures:
  
    1) Crystal::Command::FormatCommand formats stdin (bug + show-backtrace)
  
  Error running at_exit handler: Nil assertion failed
  make: *** [Makefile:89: compiler_spec] Error 1
building '/nix/store/z233pynm2vy59i8w2qdpxc3pn044jx7v-crystal-0.35.0-1-darwin-x86_64.tar.gz.drv'...
cannot build derivation '/nix/store/qmwscb3xyla6awyjhnc7jhyb8638s6aw-shards-0.11.1.drv': 1 dependencies couldn't be built
building '/nix/store/p1f1m8znhpn5r3nxbc1wr0brxgn0kjbx-crystal-0.36.1-1-darwin-x86_64.tar.gz.drv'...
building '/nix/store/lcxi16xnwa4cxnggs7nfzx8yhalr426v-crystal-binary-0.35.0.drv'...
building '/nix/store/ddjgznpvyfz64xxmqclhng7nfrlih34d-crystal-binary-0.36.1.drv'...
building '/nix/store/sgvn66abiz7wb79cbxmj9xlpgfz3f9aq-crystal-0.35.0.drv'...
builder for '/nix/store/sgvn66abiz7wb79cbxmj9xlpgfz3f9aq-crystal-0.35.0.drv' failed with exit code 2; last 10 log lines:
    Semantic: instance var doesn't infer type to be nilable if using self.class in call in assign
    Semantic: pointer allows using pointer with subclass
    Semantic: primitives types pointer of int
  
  Failures:
  
    1) Crystal::Command::FormatCommand formats stdin (bug + show-backtrace)
  
  Error running at_exit handler: Nil assertion failed
  make: *** [Makefile:92: compiler_spec] Error 1
building '/nix/store/ha7wgknabd1m5ddd2fhvl7hzpwqlksm3-crystal-0.36.1.drv'...
building '/nix/store/9v24ykzsfws0jhm8zjpasrv2ahfmcbsv-shards-0.14.1.drv'...
building '/nix/store/xh530xzb6k4qry53nj5asmifmcylsx55-ameba-0.14.3.drv'...
building '/nix/store/wsdnslszn4ai9ys395x27pc2qijp5q8v-crystal2nix-0.1.1.drv'...
building '/nix/store/nmgns3aamm4pqaq96phack7317z8wqq9-icr-unstable-2021-03-14.drv'...
building '/nix/store/ad6gjgs6sjj7g2kldnddl9qann1dj0h4-mint-0.14.0.drv'...
builder for '/nix/store/ad6gjgs6sjj7g2kldnddl9qann1dj0h4-mint-0.14.0.drv' failed with exit code 126; last 10 log lines:
    init            # Initializes a new project
    install         # Installs dependencies
    lint            # Lints the project for syntax and type errors
    loc             # Counts Lines of Code
    ls              # Language Server
    sandbox-server  # Server for compiling sandbox applications
    start           # Starts the development server
    test            # Runs the tests
    version         # Shows version
  /nix/store/3yq0m0kprzf172ljfwdl6sidvf6v6j9i-stdenv-darwin/setup: line 1323: /nix/store/s1f83gkvqi3yf1ffvvn4dx6aybjix8dx-mint-0.14.0/bin/mint.dwarf: cannot execute binary file: Exec format error
building '/nix/store/m89mp37yg5lipm38rbq1s5lf17sl20dg-oq-1.2.0.drv'...
builder for '/nix/store/m89mp37yg5lipm38rbq1s5lf17sl20dg-oq-1.2.0.drv' failed with exit code 126; last 10 log lines:
    --rawfile a f    set variable $a to a string consisting of the contents of <f>;
    --args           remaining arguments are string arguments, not files;
    --jsonargs       remaining arguments are JSON arguments, not files;
    --               terminates argument processing;
  
  Named arguments are also available as $ARGS.named[], while
  positional arguments are available as $ARGS.positional[].
  
  See the manpage for more options.
  /nix/store/3yq0m0kprzf172ljfwdl6sidvf6v6j9i-stdenv-darwin/setup: line 1323: /nix/store/kmbnj1mxv92dscdzihrz42ysgn8v062d-oq-1.2.0/bin/oq.dwarf: cannot execute binary file: Exec format error
building '/nix/store/1za9wmp00lkz903dpiy4dmg0y1aaf4kn-scry-0.9.1.drv'...
cannot build derivation '/nix/store/dl64a03dxsvss2zncpjdi4qddzs1vwsw-review-shell.drv': 5 dependencies couldn't be built
error: build of '/nix/store/dl64a03dxsvss2zncpjdi4qddzs1vwsw-review-shell.drv' failed

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/131808

1 package marked as broken and skipped:
thicket

5 packages failed to build:
crystal_0_34 crystal_0_35 mint oq shards_0_11

8 packages built:
ameba crystal crystal2nix crystal_0_36 icr lucky-cli scry shards

warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
$ nix-shell /Users/logan/.cache/nixpkgs-review/pr-131808-2/shell.nix

[nix-shell:~/.cache/nixpkgs-review/pr-131808-2]
```

--- a/pkgs/applications/version-management/git-and-tools/thicket/default.nix
+++ b/pkgs/applications/version-management/git-and-tools/thicket/default.nix
@@ -1,10 +1,10 @@
 { lib
 , fetchFromGitHub
-, crystal_0_33
+, crystal_1_0
 }:
 
 let
-  crystal = crystal_0_33;
+  crystal = crystal_1_0;
 
 in crystal.buildCrystalPackage rec {
   pname = "thicket";

--- a/pkgs/applications/version-management/git-and-tools/thicket/shards.nix
+++ b/pkgs/applications/version-management/git-and-tools/thicket/shards.nix
@@ -2,7 +2,7 @@
   ameba = {
     owner = "veelenga";
     repo = "ameba";
-    rev = "v0.10.0";
-    sha256 = "1yjxzwdhigsyjn0qp362jkj85qvg4dsyzal00pgr1srnh2xry912";
+    rev = "v0.14.3";
+    sha256 = "1cfr95xi6hsyxw1wlrh571hc775xhwmssk3k14i8b7dgbwfmm5x1";
   };
 }

--- a/pkgs/development/compilers/crystal/build-package.nix
+++ b/pkgs/development/compilers/crystal/build-package.nix
@@ -120,7 +120,7 @@ stdenv.mkDerivation (mkDerivationArgs // {
 
   installCheckPhase = args.installCheckPhase or ''
     for f in $out/bin/*; do
-      $f --help
+      $f --help > /dev/null
     done
   '';
 

--- a/pkgs/development/compilers/crystal/build-package.nix
+++ b/pkgs/development/compilers/crystal/build-package.nix
@@ -10,7 +10,7 @@
 , format ? "make"
 , installManPages ? true
   # Specify binaries to build in the form { foo.src = "src/foo.cr"; }
-  # The default `crystal build` options can be overridden with { foo.options = [ "--no-debug" ]; }
+  # The default `crystal build` options can be overridden with { foo.options = [ "--optionname" ]; }
 , crystalBinaries ? { }
 , ...
 }@args:
@@ -32,8 +32,7 @@ let
     })
     (import shardsFile));
 
-  # we previously had --no-debug here but that is not recommended by upstream
-  defaultOptions = [ "--release" "--progress" "--verbose" ];
+  defaultOptions = [ "--release" "--progress" "--verbose" "--no-debug" ];
 
   buildDirectly = shardsFile == null || crystalBinaries != { };
 

--- a/pkgs/development/compilers/crystal/default.nix
+++ b/pkgs/development/compilers/crystal/default.nix
@@ -219,48 +219,6 @@ let
 
 in
 rec {
-  binaryCrystal_0_31 = genericBinary {
-    version = "0.31.1";
-    sha256s = {
-      x86_64-linux = "0r8salf572xrnr4m6ll9q5hz6jj8q7ff1rljlhmqb1r26a8mi2ih";
-      i686-linux = "0hridnis5vvrswflx0q67xfg5hryhz6ivlwrb9n4pryj5d1gwjrr";
-      x86_64-darwin = "1dgxgv0s3swkc5cwawzgpbc6bcd2nx4hjxc7iw2h907y1vgmbipz";
-    };
-  };
-
-  crystal_0_31 = generic {
-    version = "0.31.1";
-    sha256 = "1dswxa32w16gnc6yjym12xj7ibg0g6zk3ngvl76lwdjqb1h6lwz8";
-    doCheck = false; # 5 checks are failing now
-    binary = binaryCrystal_0_31;
-  };
-
-  crystal_0_32 = generic {
-    version = "0.32.1";
-    sha256 = "120ndi3nhh2r52hjvhwfb49cdggr1bzdq6b8xg7irzavhjinfza6";
-    binary = crystal_0_31;
-  };
-
-  crystal_0_33 = generic {
-    version = "0.33.0";
-    sha256 = "1zg0qixcws81s083wrh54hp83ng2pa8iyyafaha55mzrh8293jbi";
-    binary = crystal_0_32;
-  };
-
-  crystal_0_34 = generic {
-    version = "0.34.0";
-    sha256 = "110lfpxk9jnqyznbfnilys65ixj5sdmy8pvvnlhqhc3ccvrlnmq4";
-    binary = crystal_0_33;
-  };
-
-  crystal_0_35 = generic {
-    version = "0.35.1";
-    sha256 = "0p51bjl1nsvwsm64lqq421dcsxa201w7wwq8plw4r8wqarpq0g69";
-    binary = crystal_0_34;
-    # Needs git to build as per https://github.com/crystal-lang/crystal/issues/9789
-    extraBuildInputs = [ git ];
-  };
-
   binaryCrystal_0_36 = genericBinary {
     version = "0.36.1";
     sha256s = {

--- a/pkgs/development/compilers/crystal/default.nix
+++ b/pkgs/development/compilers/crystal/default.nix
@@ -56,6 +56,7 @@ let
       buildCommand = ''
         mkdir -p $out
         tar --strip-components=1 -C $out -xf ${src}
+        patchShebangs $out/bin/crystal
       '';
     };
 
@@ -93,6 +94,10 @@ let
       outputs = [ "out" "lib" "bin" ];
 
       postPatch = ''
+        export TMP=$(mktemp -d)
+        export HOME=$TMP
+        mkdir -p $HOME/test
+
         # Add dependency of crystal to docs to avoid issue on flag changes between releases
         # https://github.com/crystal-lang/crystal/pull/8792#issuecomment-614004782
         substituteInPlace Makefile \
@@ -103,39 +108,35 @@ let
 
         ln -sf spec/compiler spec/std
 
-        # Dirty fix for when no sandboxing is enabled
-        rm -rf /tmp/crystal
-        mkdir -p /tmp/crystal
+        mkdir -p $TMP/crystal
 
         substituteInPlace spec/std/file_spec.cr \
           --replace '/bin/ls' '${coreutils}/bin/ls' \
-          --replace '/usr/share' '/tmp/crystal' \
-          --replace '/usr' '/tmp'
+          --replace '/usr/share' "$TMP/crystal" \
+          --replace '/usr' "$TMP" \
+          --replace '/tmp' "$TMP"
 
         substituteInPlace spec/std/process_spec.cr \
           --replace '/bin/cat' '${coreutils}/bin/cat' \
           --replace '/bin/ls' '${coreutils}/bin/ls' \
           --replace '/usr/bin/env' '${coreutils}/bin/env' \
           --replace '"env"' '"${coreutils}/bin/env"' \
-          --replace '"/usr"' '"/tmp"'
-
-        substituteInPlace spec/std/socket/tcp_server_spec.cr \
-          --replace '{% if flag?(:gnu) %}"listen: "{% else %}"bind: "{% end %}' '"bind: "'
+          --replace '/usr' "$TMP" \
+          --replace '/tmp' "$TMP"
 
         substituteInPlace spec/std/system_spec.cr \
           --replace '`hostname`' '`${hostname}/bin/hostname`'
 
-        # See https://github.com/crystal-lang/crystal/pull/8640
-        substituteInPlace spec/std/http/cookie_spec.cr \
-          --replace '01 Jan 2020' '01 Jan #{Time.utc.year + 2}'
-
         # See https://github.com/crystal-lang/crystal/issues/8629
         substituteInPlace spec/std/socket/udp_socket_spec.cr \
           --replace 'it "joins and transmits to multicast groups"' 'pending "joins and transmits to multicast groups"'
+      '';
 
-        # See https://github.com/crystal-lang/crystal/pull/8699
-        substituteInPlace spec/std/xml/xml_spec.cr \
-          --replace 'it "handles errors"' 'pending "handles errors"'
+      # Defaults are 4
+      preBuild = ''
+        export CRYSTAL_WORKERS=$NIX_BUILD_CORES
+        export threads=$NIX_BUILD_CORES
+        export CRYSTAL_CACHE_DIR=$TMP
       '';
 
       buildInputs = commonBuildInputs extraBuildInputs;
@@ -197,9 +198,6 @@ let
       checkTarget = "compiler_spec";
 
       preCheck = ''
-        export HOME=/tmp
-        mkdir -p $HOME/test
-
         export LIBRARY_PATH=${lib.makeLibraryPath checkInputs}:$LIBRARY_PATH
         export PATH=${lib.makeBinPath checkInputs}:$PATH
       '';

--- a/pkgs/development/compilers/crystal/default.nix
+++ b/pkgs/development/compilers/crystal/default.nix
@@ -214,8 +214,7 @@ let
         license = licenses.asl20;
         maintainers = with maintainers; [ david50407 fabianhjr manveru peterhoeg ];
         platforms = builtins.attrNames archs;
-        # Error running at_exit handler: Nil assertion failed
-        broken = lib.versions.minor version == "32" && stdenv.isDarwin;
+        broken = lib.versionOlder version "0.36.1" && stdenv.isDarwin;
       };
     })
   );
@@ -264,10 +263,19 @@ rec {
     extraBuildInputs = [ git ];
   };
 
+  binaryCrystal_0_36 = genericBinary {
+    version = "0.36.1";
+    sha256s = {
+      x86_64-linux = "065vzq34g7hgzl2mrzy9gwwsfikc78nj7xxsbrk67r6rz0a7bk1q";
+      i686-linux = "18m4b1lnd682i5ygbg6cljqjny60nn2mlrzrk765h2ip6fljqbm1";
+      x86_64-darwin = "0xggayk92zh64pb5sz77n12hkcd1hg8kw90z7gb18594q551sf1v";
+    };
+  };
+
   crystal_0_36 = generic {
     version = "0.36.1";
     sha256 = "sha256-5rjrvwZKM4lHpmxLyUVbi0Zw98xT+iJKonxwfUwS/Wk=";
-    binary = crystal_0_35;
+    binary = binaryCrystal_0_36;
   };
 
   crystal_1_0 = generic {

--- a/pkgs/development/tools/build-managers/shards/default.nix
+++ b/pkgs/development/tools/build-managers/shards/default.nix
@@ -2,6 +2,7 @@
 , fetchFromGitHub
 , crystal_0_34
 , crystal_0_36
+, crystal_1_0
 }:
 let
   generic =
@@ -49,5 +50,11 @@ rec {
     crystal = crystal_0_36;
   };
 
-  shards = shards_0_14;
+  shards_0_15 = generic {
+    version = "0.15.0";
+    sha256 = "sha256-/C6whh5RbTBkFWqpn0GqyVe0opbrklm8xPv5MIG99VU=";
+    crystal = crystal_1_0;
+  };
+
+  shards = shards_0_15;
 }

--- a/pkgs/development/tools/build-managers/shards/default.nix
+++ b/pkgs/development/tools/build-managers/shards/default.nix
@@ -1,7 +1,5 @@
 { lib
 , fetchFromGitHub
-, crystal_0_34
-, crystal_0_36
 , crystal_1_0
 }:
 let
@@ -37,18 +35,6 @@ let
 
 in
 rec {
-  # needed for anything that requires the old v1 shards format
-  shards_0_11 = generic {
-    version = "0.11.1";
-    sha256 = "05qnhc23xbmicdl4fwyxfpcvd8jq4inzh6v7jsjjw4n76vzb1f71";
-    crystal = crystal_0_34;
-  };
-
-  shards_0_14 = generic {
-    version = "0.14.1";
-    sha256 = "sha256-/C6whh5RbTBkFWqpn0GqyVe0opbrklm8xPv5MIG99VU=";
-    crystal = crystal_0_36;
-  };
 
   shards_0_15 = generic {
     version = "0.15.0";

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -11118,9 +11118,6 @@ with pkgs;
   inherit (callPackages ../development/compilers/crystal {
     llvmPackages = llvmPackages_10;
   })
-    crystal_0_33
-    crystal_0_34
-    crystal_0_35
     crystal_0_36
     crystal_1_0
     crystal;
@@ -14721,8 +14718,6 @@ with pkgs;
   shallot = callPackage ../tools/misc/shallot { };
 
   inherit (callPackage ../development/tools/build-managers/shards { })
-    shards_0_11
-    shards_0_14
     shards_0_15
     shards;
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -14723,6 +14723,7 @@ with pkgs;
   inherit (callPackage ../development/tools/build-managers/shards { })
     shards_0_11
     shards_0_14
+    shards_0_15
     shards;
 
   shellcheck = callPackage ../development/tools/shellcheck {};

